docs/plugins/npe2_manifest_specification.md

```diff
@@ -0,0 +1,606 @@
+```
+
+Readers, writers, sample data providers and widget providers refer to callable
+python functions that a plugin defines. Each callable is identified with an
+entry in the list of `commands` via a unique id. For more see {ref}`Commands`
+below.
+
+```{note}
+Python package metadata (`setup.py` or `setup.cfg`) may be used to populate
+some internal properties of the plugin manifest. These include the authors,
+description, and version.
+```
+
+(top-level-props)=
+
+## Top-level properties
+
+### Required
+
+- **name** The name of the plugin. Example: `napari-svg`. Should be a
+  [PEP-8]-compatible package name. If missing, this is populated from the
+  python package [name][setup-name].
+
+### Optional
+
+- **engine**: A SemVer compatible version string matching the versions of the
+  plugin engine that the extension is compatible with. Example: "0.1.0".
+  See [][compatibility] for details.
+- **display_name**: User-facing text to display as the name of this plugin.
+  Example: `napari SVG`. Must be 3-40 characters long, containing
+  printable word characters, and must not begin or end with an underscore,
+  white space, or non-word character.
+- **entry_point**: The module containing the `activate()` function. Example:
+  `foo.bar.baz`. This should be a fully qualified module string.
+
+```{admonition} activation and deactivation
+An `activate()` function can be used to perform one-time initialization for a
+plugin.  Similarly, a `deactivate()` function can be used to perform tear-down
+on deactivation.  These must be defined relative to the `entry_point`.
+```
+
+(compatibility)=
+
+### Compatibility
+
+A goal of `npe2` was to make it possible to add or change the plugin interface
+without breaking existing plugins. This is achieved by defining an
+**engine version**. When we want to change the plugin interface, we increment
+this number to indicate there is something new. New plugins can opt-in to the
+new behavior by declaring compatibility with that version.
+
+When authoring a plugin, you will need to describe what is the plugin's
+compatibility with `npe2`. This can be done via the `engine` field in the
+plugin manifest (see {ref}`top-level-props`).
+
+You can use the `engine` field to makes sure the plugin only gets installed
+for clients that use the plugin specification you depend on.
+
+For example, imagine you develop your plugin using the `0.1.0` engine
+specification. A few months of development go by and a new API is introduced
+to the plugin engine defining a new _engine version_, `0.2.0`. You add that
+new API and update your plugin's manifest file accordingly.
+
+Users running older versions of napari will not have the new API. Their plugin
+engine will inspect the _engine version_ declared by your plugin and correctly
+prevent your plugin from being used. When those users upgrade their napari
+installation, they'll be able to take advantage of the new functionality you
+added.
+
+(commands)=
+
+## Commands
```

Comment:
I'd really like to see almost all of this file be pulled from the `Field(description=...)` and the class docstrings of the npe2 manifest.  Otherwise, we'll always have an issue with knowing whether they're in sync.  If you want some tips for that, let me know

Reply:
I would love tips!